test/test_sparse.py

```diff
@@ -979,15 +979,90 @@ def test_factory_size_check(self):
         indices = self.IndexTensor([[1, 2], [0, 2]])
         values = self.ValueTensor([[1, 1, 1], [1, 1, 1]])
         sizes = torch.Size([3, 3, 2])
-        with self.assertRaisesRegex(RuntimeError, "values and sizes are inconsistent"):
+        with self.assertRaisesRegex(RuntimeError, "values has incorrect size"):
             self.SparseTensor(indices, values, sizes)
 
+    def test_factory_default(self):
+        tensor = self.SparseTensor()
+        expected_indices = self.IndexTensor(1, 0)
+        expected_size = torch.Size([0])
+        self.assertEqual(tensor._indices(), expected_indices)
+        self.assertEqual(tensor.shape, expected_size)
+
     def test_factory_empty_indices(self):
         device = 'cuda' if self.is_cuda else 'cpu'
-        tensor = torch.sparse_coo_tensor([], [], torch.Size([]), device=device)
-        expected_indices = torch.tensor([], dtype=torch.long, device=device)
+        tensor = self.SparseTensor()
+        expected_indices = torch.empty((1, 0), dtype=torch.long, device=device)
+        self.assertEqual(tensor._indices(), expected_indices)
+
+        tensor = torch.sparse_coo_tensor(torch.Size([2, 0]), device=device)
+        expected_indices = torch.empty((2, 0), dtype=torch.long, device=device)
         self.assertEqual(tensor._indices(), expected_indices)
 
+        tensor = torch.sparse_coo_tensor(torch.Size([2, 2, 0]), device=device)
+        expected_indices = torch.empty((3, 0), dtype=torch.long, device=device)
+        self.assertEqual(tensor._indices(), expected_indices)
+
+        tensor = torch.sparse_coo_tensor(torch.Size([2, 2, 0, 0]), device=device)
+        expected_indices = torch.empty((4, 0), dtype=torch.long, device=device)
+        self.assertEqual(tensor._indices(), expected_indices)
+
+    def test_factory_nnz(self):
+        indices = self.IndexTensor([[0]])  # (sparseDims, nnz): (1, 1)
+        values = self.ValueTensor([[1, 1], [1, 1]])  # (nnz, ...): (2, 2)
+        sizes = torch.Size([2, 2])
+        with self.assertRaisesRegex(RuntimeError, "indices and values must have same nnz"):
+            self.SparseTensor(indices, values, sizes)
+
+    def test_factory_zero_nnz(self):
+        device = 'cuda' if self.is_cuda else 'cpu'
+        t = torch.sparse_coo_tensor(torch.empty(1, 0), torch.empty(0, 2, 4, 0), device=device)
```

Comment:
I thought you had to provide the size for cuda sparse tensors -- is that no longer the case?

Reply:
I think now it can work when only `indices` and `values` are provided, and it will infer the size from them.